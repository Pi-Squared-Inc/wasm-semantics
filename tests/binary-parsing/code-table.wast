(module $mymodule
  (table 2 funcref)
  (table 2 funcref)
  (elem (i32.const 0) 0)
  (elem (i32.const 0) 0)
  (func (param i32) (result i32)
    (i32.const 1)
    (i32.const 1)
    (table.get 0)
    (table.set 0)

    (table.size 0)
    drop

    (i32.const 1)
    (table.get 0)
    (i32.const 10)
    (table.grow 0)
    drop

    (i32.const 10)
    (i32.const 1)
    (table.get 0)
    (i32.const 10)
    (table.fill 0)

    (i32.const 1)
    (i32.const 1)
    (i32.const 1)
    (table.copy 0 1)

    (i32.const 1)
    (i32.const 1)
    (i32.const 1)
    (table.init 0 1)

    (elem.drop 0)
    (return (i32.const 1))
  )
)
