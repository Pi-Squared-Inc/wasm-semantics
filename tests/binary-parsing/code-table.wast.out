<generatedTop>
  <k>
    #module (... types: #type (... type: [ i32  .ValTypes ] -> [ i32  .ValTypes ] , metadata:  )  .EmptyStmts , funcs: #func (... type: 0 , locals: [ .ValTypes ] , body: i32 . const 1  i32 . const 1  #table.get ( 0 )  #table.set ( 0 )  #table.size ( 0 )  drop  i32 . const 1  #table.get ( 0 )  i32 . const 10  #table.grow ( 0 )  drop  i32 . const 10  i32 . const 1  #table.get ( 0 )  i32 . const 10  #table.fill ( 0 )  i32 . const 1  i32 . const 1  i32 . const 1  #table.copy ( 0 , 1 )  i32 . const 1  i32 . const 1  i32 . const 1  #table.init ( 0 , 1 )  #elem.drop ( 0 )  i32 . const 1  return  .EmptyStmts , metadata: #meta (... id:  , localIds: .Map ) )  .EmptyStmts , tables: #table (... limits: #limitsMin ( 2 ) , type: funcref , metadata:  )  #table (... limits: #limitsMin ( 2 ) , type: funcref , metadata:  )  .EmptyStmts , mems: .EmptyStmts , globals: .EmptyStmts , elem: #elem (... type: funcref , elemSegment: ListItem ( < funcref > 0 ) , mode: #elemActive (... table: 0 , offset: i32 . const 0  .EmptyStmts ) , oid:  )  #elem (... type: funcref , elemSegment: ListItem ( < funcref > 0 ) , mode: #elemActive (... table: 0 , offset: i32 . const 0  .EmptyStmts ) , oid:  )  .EmptyStmts , data: .EmptyStmts , start: .EmptyStmts , importDefns: .EmptyStmts , exports: .EmptyStmts , metadata: #meta (... id:  , funcIds: .Map , filename: .String ) ) ~> .K
  </k>
  <wasm>
    <instrs>
      .K
    </instrs>
    <valstack>
      .ValStack
    </valstack>
    <curFrame>
      <locals>
        .List
      </locals>
      <curModIdx>
        .Int
      </curModIdx>
    </curFrame>
    <moduleRegistry>
      .Map
    </moduleRegistry>
    <moduleIds>
      .Map
    </moduleIds>
    <moduleInstances>
      .ModuleInstCellMap
    </moduleInstances>
    <nextModuleIdx>
      0
    </nextModuleIdx>
    <mainStore>
      <funcs>
        .FuncDefCellMap
      </funcs>
      <nextFuncAddr>
        0
      </nextFuncAddr>
      <tabs>
        .TabInstCellMap
      </tabs>
      <nextTabAddr>
        0
      </nextTabAddr>
      <mems>
        .List
      </mems>
      <globals>
        .GlobalInstCellMap
      </globals>
      <nextGlobAddr>
        0
      </nextGlobAddr>
      <elems>
        .ElemInstCellMap
      </elems>
      <nextElemAddr>
        0
      </nextElemAddr>
    </mainStore>
    <deterministicMemoryGrowth>
      true
    </deterministicMemoryGrowth>
  </wasm>
</generatedTop>
